pkg/skaffold/deploy/kubectl/cli.go

```diff
@@ -58,28 +58,29 @@ func (c *CLI) Apply(ctx context.Context, out io.Writer, manifests ManifestList)
 	if len(updated) == 0 {
 		return nil, nil
 	}
-
-	buf := bytes.NewBuffer([]byte{})
-	writer := bufio.NewWriter(buf)
-	if err := c.Run(ctx, manifests.Reader(), writer, "apply", c.Flags.Apply, "-f", "-"); err != nil {
-		if !strings.Contains(buf.String(), "field is immutable") {
-			return nil, err
-		}
-		// If the output contains the string 'field is immutable', we want to delete the object and recreate it
-		// See Issue #891 for more information
-		if err := c.Delete(ctx, out, manifests); err != nil {
-			return nil, errors.Wrap(err, "deleting manifest")
-		}
-		if err := c.Run(ctx, manifests.Reader(), out, "apply", c.Flags.Apply, "-f", "-"); err != nil {
-			return nil, errors.Wrap(err, "kubectl apply after deletion")
-		}
-	} else {
-		// Write output to out
-		if _, err := out.Write(buf.Bytes()); err != nil {
-			return nil, errors.Wrap(err, "writing to out")
+	for _, mfst := range manifests {
+		buf := bytes.NewBuffer([]byte{})
+		writer := bufio.NewWriter(buf)
+		ml := ManifestList{mfst}
```

Comment:
no need to convert this back to a manifest list, mfst is just a []byte at this point